firebase-perf/src/main/java/com/google/firebase/perf/application/AppStateUpdateHandler.java

```diff
@@ -31,6 +31,7 @@ public abstract class AppStateUpdateHandler implements AppStateCallback {
   // It must be a weak reference because unregisterForAppState() is called typically from
   // Trace.stop() and user may forget to call Trace.stop(), if it was a strong reference,
   // the registration in AppStateMonitor will prevent Trace to be deallocated.
+  /** @hide */
```

Comment:
Why is a `private final` var need the hide annotation? I think we can document it.

Reply:
Dackka currently leaks `private final` variables with public getters. And if you look at the public getter for this- it's hidden as well. This isn't supposed to be in the public docs, it's only `@VisibleForTesting`.

See [b/241259955](https://b.corp.google.com/issues/241259955) for context.

Reply:
If you feel strongly about it, I'm down for it. But I do weigh more towards not doing it.

If you take a look at [this commit](https://github.com/firebase/firebase-android-sdk/pull/4714/commits/f4ea4f130ecaa8ef2420e812972549cdee28c651), there's a considerable amount of changes that originated from a single bug ([go/dokka-upstream-bug/2665](http://go/dokka-upstream-bug/2665)).

Would you also suggest we add a `TODO()` at the site of each one? I think leaving as is should be fine- as it doesn't cause any issues. Should someone come across it and change it, if we haven't updated Dackka to a version that fixes it then they'll see the issue as soon as they try to submit a PR and it notes the Javadoc changes. Otherwise, it won't cause any harm.

Thoughts?

Reply:
FWIW the changes in that commit, while necessary to make dackka work, do not jump as weird the same way as annotating a private field `@hide`. That being said, I don't feel strongly about it, so feel free to leave it as is.